packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/tool-list/PaletteToolList.tsx

```diff
@@ -82,11 +82,19 @@ export const PaletteToolList = ({ palette, onToolClick }: PaletteToolListProps)
 
   const onBackToMainList = () => {
     setState((prevState) => ({ ...prevState, toolSection: null }));
+    document.getElementById('palette')?.focus();
```

Comment:
Use a proper API, this child component here is knowing the internal structure of the parent component in which it is currently being used. We don't want this kind of coupling.

Reply:
If you are relying on an identifier in our dom code to do something, you are missing something in term of API or lifecycle. We don't need any id in our code to make it work. I would argue that relying on Reactflow ids may not be perfect too but that's another issue.

Reply:
Done